x-pack/plugin/esql/src/main/java/org/elasticsearch/xpack/esql/parser/IdentifierBuilder.java

```diff
@@ -51,29 +52,51 @@ protected static String quoteIdString(String unquotedString) {
         return "`" + unquotedString.replace("`", "``") + "`";
     }
 
+    @Override
+    public String visitClusterString(EsqlBaseParser.ClusterStringContext ctx) {
+        if (ctx == null) {
+            return null;
+        } else if (ctx.UNQUOTED_SOURCE() != null) {
+            return ctx.UNQUOTED_SOURCE().getText();
+        } else {
+            return unquote(ctx.QUOTED_STRING().getText());
+        }
```

Comment:
See my comment in the grammar - this method can then be either remove or delegate to visitIndexString.